### PR TITLE
Tutorial

### PR DESCRIPTION
**Pull request by WyriHaximus**
- [X] Connecting
- [x] Publish
- [x] Subscribe
- [x] Get a single message
- [x] pre-fetch count
- [x] Async notes

### CODE REVIEW

**Comment by WyriHaximus** [comment]
Ok done writing the tutorial unless there are any suggestions of items to add :)

File: README.md [hunk 1]
```diff
@@ -66,7 +66,116 @@ $ php benchmark/producer.php N & php benchmark/consumer.php
 
 ## Tutorial
 
-TODO, see `benchmarks/` for basic use.
+### Connecting
+
+When instanciating the bunny `Client` accepts an array with connection options:
```
Comment (jakubkulhan) [inline README.md]: `instanciating` -> `instantiating`
`bunny` -> `BunnyPHP`

File: README.md [hunk 2]
```diff
@@ -66,7 +66,116 @@ $ php benchmark/producer.php N & php benchmark/consumer.php
...
+
+```php
+$connection = [
+    'host'      => 'HOSTNAME',
+    'vhost'     => 'VHOST', // The default vhost is /
+    'user'      => 'USERNAME',
```
Comment (jakubkulhan) [inline README.md]: default is `guest`

File: README.md [hunk 3]
```diff
@@ -66,7 +66,116 @@ $ php benchmark/producer.php N & php benchmark/consumer.php
...
+    'password'  => 'PASSWORD',
```
Comment (jakubkulhan) [inline README.md]: default is `guest`

File: README.md [hunk 4]
```diff
@@ -66,7 +66,116 @@ $ php benchmark/producer.php N & php benchmark/consumer.php
...
+    $message,    // The message you're publishing as a string
+    [],          // Any headers you want to add to the message
+    '',          // Exchange name
+    'queue_name' // Routing key, in this example the queue's name
+);
+```
+
+### Subscribing to a queue
+
+Subscribing to a queue can be done in two ways. The first way will run indefinetly:
```
Comment (jakubkulhan) [inline README.md]: `indefinetly` -> `indefinitely`

File: README.md [hunk 5]
```diff
@@ -66,7 +66,116 @@ $ php benchmark/producer.php N & php benchmark/consumer.php
...
+$message = $channel->get('queue_name');
+
+// Handle message
+
+$channel->ack($message); // Acknowledge message
+```
+
+### Prefetch count
+
+A way to control how many messages are prefetched by bunny when consuming a queue is by using the channel's QOS method. In the example below only 5 messages will be prefetched. Combined with acknowledging messages this turns into an effetive flow controll for your applications. (Especially asynchronous applications.) No new messages will be fetched unless one has been acknowledged.
```
Comment (jakubkulhan) [inline README.md]: `bunny` -> `BunnyPHP`

**Comment by jakubkulhan** [comment]
Great work! I've found some typos, please correct them before merge. Thank you.

**Comment by WyriHaximus** [comment]
Thanks and done :+1:

**Pull request merged**